csrc/scheduler/mma_utils.cpp

```diff
@@ -1247,6 +1249,154 @@ void canonicalizeMmaTvOrdering(TensorView* tv) {
   tv->reorder(order_map);
 }
 
+ProblemIterDomainsOpt getProblemIterDomains(Fusion* fusion) {
+  auto mma_exprs = ir_utils::getMmaOps(fusion);
+  if (mma_exprs.size() != 1) {
+    std::stringstream ss;
+    ss << "Invalid number of MmaOp instances in fusion, expected 1, got "
+       << mma_exprs.size();
+    return ss.str();
+  }
+  const auto mma_output = mma_exprs.front()->out();
+
+  // NOTE: the iter domains of MMA output should be [...,M,K,N]
+  IterDomain* m = nullptr;
+  IterDomain* n = nullptr;
+  IterDomain* k = nullptr;
+
+  const auto leaf_domains =
+      static_cast<const TensorView*>(mma_output)->getLeafDomain();
+  const auto concrete =
+      TensorDomain::noReductions(TensorDomain::noBroadcasts(leaf_domains));
+  if (concrete.size() < MIN_MATMUL_INPUTS_NUMBER) {
+    std::stringstream ss;
+    ss << "Failed to find the minimum number of MMA input candidates, expected "
+       << MIN_MATMUL_INPUTS_NUMBER << ", got " << concrete.size();
+    return ss.str();
+  }
+
+  // M,N are inner most concrete iter domains
+  m = concrete.rbegin()[1];
+  n = concrete.rbegin()[0];
+
+  // K is a reduction domain, search for the inner most reduction domain
+  for (auto iter_domain = leaf_domains.rbegin();
+       iter_domain != leaf_domains.rend();
+       ++iter_domain) {
+    if ((*iter_domain)->isReduction()) {
+      k = *iter_domain;
+      break;
+    }
+  }
+  TORCH_INTERNAL_ASSERT(k != nullptr, "Failed to find K domain in MMA output");
+
+  return ProblemIterDomains{m, n, k};
+}
+
+MatmulProblemLayoutOpt getMatmulLayout(Fusion* fusion) {
+  using MmaDomains = MmaOptions::MmaDomains;
+  using DomainsDesc = std::vector<MmaDomains>;
+  using DependenciesMap = std::map<TensorView*, DomainsDesc>;
+
+  ComputeAtMap ca_map(fusion);
+  const auto mma_input_candidates =
+      ir_utils::filterByType<TensorView>(fusion->inputs()).vector();
+  if (mma_input_candidates.empty()) {
+    return {"Failed to find any TV that is fusion input"};
+  }
+
+  const auto mma_output_domains = getProblemIterDomains(fusion);
+  if (!mma_output_domains.isValid()) {
+    return mma_output_domains.getErrorMsg();
+  }
+
+  const auto [m, n, k] = mma_output_domains.getData();
+
+  DependenciesMap deps_map;
+
+  // NOTE: MMA input requirements:
+  // - 2 or more domains
+  // - no broadcasts
+  // - reductions do not matter
+  for (const auto in_tv : mma_input_candidates) {
+    if (in_tv->nDims() < 2) {
+      continue;
+    }
+    if (in_tv->hasBroadcast()) {
```

Comment:
Also, we are not checking that there isn't any batch dimension, because we are thinking about support it in the future. But why disallowing broadcast? A batch dimension can be a broadcast.

Reply:
I see your point, I will remove these checks for now. If they are needed, we can bring them back.

As with the other change, I want to have it done in the follow up PR. Please let me know if this should be done in this PR.
Thanks!

Reply:
The suggestion has been applied in https://github.com/NVIDIA/Fuser/pull/442